### PR TITLE
Create key w better name; restrict key deletion

### PR DESCRIPTION
**Pull request by landonreed**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [ ] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

This PR adds a couple of more changes to API Key management:
- restricts key deletion to admin users -- the thinking here is that we may not want people to constantly churn through API keys. That could cause us/admins to lose track of API usage as they're deleted.
- creates keys with more descriptive names (this way it's easier to track back the key in the AWS console)

### CODE REVIEW

**Review by evansiroky — APPROVED** [review]

**Review by br648 — APPROVED** [review]
LGTM.

File: src/main/java/org/opentripplanner/middleware/utils/ApiGatewayUtils.java
Comment (binh-dam-ibigroup) [inline src/main/java/org/opentripplanner/middleware/utils/ApiGatewayUtils.java]: Perhaps add a comment explaining what happens if the usage plan does not exist. (does that trigger an exception?)

**Review by binh-dam-ibigroup — APPROVED** [review]
One additional comment clarification could be added when checking the existence of the usage plan ID, otherwise this PR is good to go.

File: src/main/java/org/opentripplanner/middleware/utils/ApiGatewayUtils.java
Comment (landonreed) [reply src/main/java/org/opentripplanner/middleware/utils/ApiGatewayUtils.java]: Yes, exactly. I added some changes to clarify the behavior (there's a new exception class as well).

**Comment by landonreed** [comment]
Made some minor tweaks to how exceptions get generated, so I thought I should get some quick feedback before merging.

**Review by evansiroky — APPROVED** [review]

File: src/main/java/org/opentripplanner/middleware/controllers/api/ApiUserController.java
Comment (binh-dam-ibigroup) [inline src/main/java/org/opentripplanner/middleware/controllers/api/ApiUserController.java:124]: As a result of this new restriction, test `ApiKeyManagementTest#canDeleteApiKeyForSelf()` should be removed.

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
Sorry, I need to change approval status. One E2E test may be obsolete, is failing as noted, and may need to be removed.

File: src/main/java/org/opentripplanner/middleware/controllers/api/ApiUserController.java
Comment (landonreed) [reply src/main/java/org/opentripplanner/middleware/controllers/api/ApiUserController.java:124]: Good catch. How about we change it to cannotDeleteApiKeyForSelf?

Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/controllers/api/ApiUserController.java:124]: That should do too, and we'll have a test to police the new restriction.

**Review by binh-dam-ibigroup — APPROVED** [review]
Looks good!

**Pull request merged**
